### PR TITLE
added history function to the calculator

### PR DESCRIPTION
**Pull request by AwetKebedom**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -7,11 +7,13 @@
         <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
         <link rel="stylesheet" media="screen" href="https://fontlibrary.org/face/segment7" type="text/css"/>
         <link href="https://fonts.googleapis.com/css?family=Rubik" rel="stylesheet">
+        
         <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjs/6.2.2/math.js"></script>
 
         <script src="script.js"></script>
     </head>
     <body>
+<!-- <<<<<<< Updated upstream
```
Comment (Tesfit12) [inline index.html]: why do you need line 16

File: script.js [hunk 2]
```diff
@@ -19,14 +27,25 @@ function zero() {
...
+// <<<<<<< Updated upstream
+//         try {a.value = math.evaluate(exp);}
+//         catch (e) {
+// =======
+        try {a.value = a.value +'=' + math.evaluate(exp);
+        histor.push(a.value)
+        hist();
+        }
+       catch (e) {
+// >>>>>>> Stashed changes
```
Comment (Tesfit12) [inline script.js]: how about this line you do not need it i think

File: script.js [hunk 3]
```diff
@@ -52,4 +71,37 @@ function clean() {
...
+        document.getElementById('result').innerHTML += '   ';
+        histor = [];
+
+        
+      
+       
+    }
+    
+}
+// >>>>>>> Stashed changes
```
Comment (Tesfit12) [inline script.js:137]: this one also

File: script.js [hunk 4]
```diff
@@ -52,4 +71,37 @@ function clean() {
 function back() {
     var exp = document.form.textview.value;
     document.form.textview.value = exp.substring(0, exp.length-1);
-}
\ No newline at end of file
+}
+// <<<<<<< Updated upstream
+// =======
```
Comment (Tesfit12) [inline script.js:106]: even this

File: script.js [hunk 5]
```diff
@@ -2,15 +2,23 @@
...
 function insert(num) {
-
+    
+    
     var a = document.form.textview;
 
     a.value += num;
+    
+    // histor += num;
+
```
Comment (dorimedini) [inline script.js]: If this is dead code, please delete these lines

**Review by Tesfit12 — CHANGES_REQUESTED** [review]

File: script.js [hunk 6]
```diff
@@ -2,15 +2,23 @@
...
     var a = document.form.textview;
 
     a.value += num;
+    
+    // histor += num;
+    
 }
 function opt(amd) {
+    
+
```
Comment (dorimedini) [inline script.js]: No need for these empty lines, right?

File: script.js [hunk 7]
```diff
@@ -2,15 +2,23 @@
...
+    
 }
 function opt(amd) {
+    
+    
     var view = document.form.textview;
     if (view.value != "") {
         view.value += amd;
+        // histor += amd;
+
```
Comment (dorimedini) [inline script.js:29]: Please delete dead code

File: script.js [hunk 8]
```diff
@@ -19,14 +27,25 @@ function zero() {
         view.value += '0';
     }
 }
+var histor = [];
 function equal() {
-    var button = document.getElementsByClassName('none');
+    
+    // var button = document.getElementsByClassName('none');
```
Comment (dorimedini) [inline script.js]: Please delete dead code

File: script.js [hunk 9]
```diff
@@ -19,14 +27,25 @@ function zero() {
...
     var exp = a.value;
+    
 
     if (exp) {
-        try {a.value = math.evaluate(exp);}
-        catch (e) {
+// <<<<<<< Updated upstream
+//         try {a.value = math.evaluate(exp);}
+//         catch (e) {
+// =======
```
Comment (dorimedini) [inline script.js:58]: Ahhhh... merge conflicts... so much fun. No need for these lines anymore :)

File: script.js [hunk 10]
```diff
@@ -19,14 +27,25 @@ function zero() {
...
+        try {a.value = a.value +'=' + math.evaluate(exp);
+        histor.push(a.value)
+        hist();
+        }
+       catch (e) {
```
Comment (dorimedini) [inline script.js]: The indentation here is a bit strange, please align the code in a nice way

File: script.js (hunk 2 above)
Comment (dorimedini) [inline script.js:64]: Delete merge-conflict artifact (we won't miss it)

File: script.js [hunk 11]
```diff
@@ -52,4 +71,37 @@ function clean() {
...
+// function sinFunc(){
+//     var a = document.form.textview.value;
+    
+//     document.form.textview.value = Math.sin(a);
+// }
+// function sqrFunc(){
+//     var a = document.form.textview.value;
+    
+//     document.form.textview.value = Math.sqrt(a);
+// }
```
Comment (dorimedini) [inline script.js:116]: Please delete dead code

File: script.js [hunk 12]
```diff
@@ -52,4 +71,37 @@ function clean() {
...
+function hist()
+{
+    var i;
+    for (i = 0; i < histor.length; i++) {
+        
+            // if (histor[i] == histor[(histor.length)-i] ){
+            //     continue;
+                
+            // }
```
Comment (dorimedini) [inline script.js:125]: Please delete dead code

File: script.js [hunk 13]
```diff
@@ -52,4 +71,37 @@ function clean() {
...
+        
+        document.getElementById('result').innerHTML += histor[i] + '<br>';
+        document.getElementById('result').innerHTML += '   ';
+        histor = [];
+
+        
+      
+
```
Comment (dorimedini) [inline script.js:133]: No need for so many new lines

File: style.css [hunk 14]
```diff
@@ -89,6 +89,22 @@ table {
...
 }
+.col-md-3
+{
+    width: 33%;
+    border: solid cadetblue 3px;
+    margin: 40px 20px;
+    background-color: darkgray;
+    /* -webkit-box-shadow: inset 7px -24px 92px -3px rgba(0,0,0,0.43);
+-moz-box-shadow: inset 7px -24px 92px -3px rgba(0,0,0,0.43);
+box-shadow: inset 7px -24px 92px -3px rgba(0,0,0,0.43); */
```
Comment (dorimedini) [inline style.css:100]: Please delete dead code

File: style.css [hunk 15]
```diff
@@ -89,6 +89,22 @@ table {
...
+box-shadow: inset 0 -5px 10px 0 #fff,
```
Comment (dorimedini) [inline style.css:101]: Fix indentation here (`box-shadow` should be one tab out)

File: style.css [hunk 16]
```diff
@@ -148,3 +164,9 @@ table {
     background: transparent;
     color: auto;
 }
+/* #result{
+    width: 5em;
+    margin-left: 50%;
+    background-color: cadetblue;
+    padding: 10px;
+} */
```
Comment (dorimedini) [inline style.css:172]: Please delete dead code

**Review by dorimedini — CHANGES_REQUESTED** [review]
Very good job resolving the nasty merge conflict... I only requested cosmetic changes

**Pull request merged**